enterprise/reporting/reporting.go

```diff
@@ -374,6 +384,19 @@ func (r *HandleT) mainLoop(ctx context.Context, clientName string) {
 	r.getMinReportedAtQueryTime = stats.Default.NewTaggedStat(StatReportingGetMinReportedAtQueryTime, stats.TimerType, tags)
 	r.getReportsQueryTime = stats.Default.NewTaggedStat(StatReportingGetReportsQueryTime, stats.TimerType, tags)
 	r.requestLatency = stats.Default.NewTaggedStat(StatReportingHttpReqLatency, stats.TimerType, tags)
+
+	lastReportedAt := time.Now().UTC().Unix() / 60
+	go func() {
+		// for montering reports pileups
+		for {
+			currentMs := time.Now().UTC().Unix() / 60
+			stats.Default.NewTaggedStat(
+				"reporting_metrics_pileup", stats.GaugeType, stats.Tags{"client": clientName},
+			).Gauge(int(currentMs - lastReportedAt))
+			time.Sleep(2 * time.Minute)
```

Comment:
Avoid time.Sleep especially for such durations of time, can cause issues with graceful shutdown and testing.

```suggestion
			select {
	    case <-ctx.Done():
				break;
			case <-time.After(2 * time.Minute):
			}
		}
			```